src/javaRosa.js

```diff
@@ -576,7 +576,12 @@ define([
 
         widget.save = function () {
             // override save to call out to rename itext
-            widget.mug.setItextId(widget.path, widget.getValue());
+            if (widget.mug.setItextId(widget.path, widget.getValue())) {
```

Comment:
The answer came to me after asking all those questions. Mug properties usually fire change events when they are set, however `setItextId` is not doing that. It may make sense to simply have it fire that event unconditionally, or maybe if there are some places (for example during parsing) where change events should not be fired, then maybe you'll want to add a third parameter where you can indicate whether it should fire a change event or not. In this case it should. With the current design of Vellum it seems like it is not the widget's job to fire internal mug change events.